src/components/calculator.js

```diff
@@ -1,41 +1,50 @@
 import React from 'react';
+import calculate from './logic/calculate.js';
 
-function Pad() {
-  return (
-    <div>
-      <ul>
-        <li className="screen"><input className="" type="text" placeholder="0" /></li>
-        <li className="pad same"><button type="submit">AC</button></li>
-        <li className="pad same"><button type="submit">+/-</button></li>
-        <li className="pad same"><button type="submit">%</button></li>
-        <li className="pad different"><button type="submit">+</button></li>
-        <li className="pad same"><button type="submit">7</button></li>
-        <li className="pad same"><button type="submit">8</button></li>
-        <li className="pad same"><button type="submit">9</button></li>
-        <li className="pad different"><button type="submit">*</button></li>
-        <li className="pad same"><button type="submit">4</button></li>
-        <li className="pad same"><button type="submit">5</button></li>
-        <li className="pad same"><button type="submit">6</button></li>
-        <li className="pad different"><button type="submit">-</button></li>
-        <li className="pad same"><button type="submit">1</button></li>
-        <li className="pad same"><button type="submit">2</button></li>
-        <li className="pad same"><button type="submit">3</button></li>
-        <li className="pad different"><button type="submit">/</button></li>
-        <li className="pad same zero"><button type="submit">0</button></li>
-        <li className="pad same"><button type="submit">.</button></li>
-        <li className="pad different"><button type="submit">=</button></li>
-      </ul>
-    </div>
-  );
-}
+export default class MakeOperations extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  eventhandler = (e) => {
+    const b = e.target.textContent;
+    const calc = calculate(this.state, b);
+    this.setState(calc);
+  };
 
-class Component extends React.Component {
   render() {
+    const buttons = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x',
+      '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
+
+    const { next, total, operation } = this.state;
+    let result = '';
+    if (total) {
+      result = `${total} ${operation || ''} ${next || ''}`;
+    } else if (next) {
+      result = next;
+    }
+
+    function addClass(button) {
+      if (button === '+' || button === 'x' || button === '-' || button === '÷' || button === '=') {
+        return 'different';
+      }
+      if (button === '0') {
+        return 'zero';
+      }
+      return '';
+    }
     return (
-      <Pad />
+      <div>
+        <ul>
+          <li className='screen'>
+              <h3>
+                {result}
+              </h3>
+            </li>
+            {buttons.map((btn) => (<li className={`pad ${addClass(btn)}`} key={btn}><button key={btn} onClick={this.eventhandler} type="button">{btn}</button></li>))}
+        </ul>
+      </div>
     );
   }
 }
-export default Component;
```

Comment:
- [ ] I had this while trying to run the project locally, to check if the project works perfectly. Please fix it.
![user-3](https://user-images.githubusercontent.com/55635977/139742425-36d9d2b9-0faf-4fef-9e2a-cebfecce152b.png)

 - [ ] To implement this milestone, you are also supposed or required to use lifecycle methods.

Reply:
fixed